neo/Network/P2P/Payloads/VersionPayload.cs

```diff
@@ -1,42 +1,53 @@
 ﻿using Neo.IO;
+using Neo.IO.Caching;
+using Neo.Network.P2P.Capabilities;
 using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 
 namespace Neo.Network.P2P.Payloads
 {
     public class VersionPayload : ISerializable
     {
+        /// <summary>
+        /// Reflection cache for ConsensusMessageType
+        /// </summary>
+        private static readonly ReflectionCache<byte> ReflectionCache = ReflectionCache<byte>.CreateFromEnum<NodeCapabilities>();
+
+        const int MaxCapabilities = 32;
+
         public uint Magic;
         public uint Version;
         public VersionServices Services;
         public uint Timestamp;
-        public ushort Port;
         public uint Nonce;
         public string UserAgent;
         public uint StartHeight;
+        public Dictionary<NodeCapabilities, INodeCapability> Capabilities;
```

Comment:
Why not use a struct:

```c#
struct NodeCapability
{
    CapabilitiyType Type;
    ushort Port;
}
```

And then:

```c#
public NodeCapability[] Capabilities;
```

Reply:
Because we can want other kind of structs, strings for example

Reply:
Maybe it could be a List, instead of a Dictionary

Reply:
List is good, or array. I think ws should be a port too, not a string.